overlord/snapshotstate/backend/backend.go

```diff
@@ -476,42 +476,53 @@ func Import(ctx context.Context, id uint64, r io.Reader) (int64, []string, error
 	// walk the cache directory to store the files
 	dir, err := osOpen(p)
 	if err != nil {
-		return 0, nil, fmt.Errorf("failed opening import cache: %v", comment)
+		return 0, nil, fmt.Errorf("failed opening import cache for %s: %v", comment, err)
 	}
 	defer dir.Close()
-	names, err := dirNames(dir, 100)
-	if err != nil {
-		return 0, nil, fmt.Errorf("failed read from import cache: %v", comment)
-	}
 
-	// move the files into place with the new local set ID
-	snapNames, err := moveCachedSnapshots(names, id, p)
+	var snapNames []string
+	var readErr error
+	for readErr == nil {
+		var names []string
+		names, readErr = dirNames(dir, 100)
+		if err != nil {
```

Comment:
Hm, what `err` is this checking? readErr? or a leftover err?

Reply:
Thanks for catching, fixed.